### PR TITLE
engine: fix a comment

### PR DESCRIPTION
**Pull request by andreimatei**
A misleading comment was suggesting that an MVCCMetadata is adjusted
when an intent is resolved. In fact, it is deleted.

Release note: None

### CODE REVIEW

**Comment by andreimatei** [comment]
Tobi I've deleted a part of a comment you wrote about intent epochs. I found it to be more confusing than useful, given that the code has no relation which the epochs being discussed.
But I'll put it back if you liked it.

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32541)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/storage/engine/mvcc.go, line 2156 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32541#-LRrxidT0k5a2e7fV5FW:-LRrxidU0D0bhfEbsDyU:bph1oee) ([raw file](https://github.com/cockroachdb/cockroach/blob/c404eed0e6cf8bbe1b935e2166c64eb012ca7025/pkg/storage/engine/mvcc.go#L2156)):*
> ```Go
> 
> 	// Otherwise, we're deleting the intent, which includes deleting the
> 	// MVCCMetadata.
> ```

I'll let @tbg weigh-in on the deletion of the epoch handling portion of this comment. The fix to not mention updating the metadata timestamp LGTM.

**Review by tbg — APPROVED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32541)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/storage/engine/mvcc.go, line 2160 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32541#-LRs1xMJ7rs17YvhNELT:-LRs1xMJ7rs17YvhNELU:ba1fsky) ([raw file](https://github.com/cockroachdb/cockroach/blob/245799c44d11966371ce1f45cf90e57c85b66bc2/pkg/storage/engine/mvcc.go#L2160)):*
> ```Go
> 	// key.
> 	//
> 	// Note that the somewhat unintuitive case of an ABORT with
> ```

Note the condition above this block. I think the comment makes sense, though perhaps it should be moved up to the `return false, nil`. The comment explains why we don't necessarily have `meta.Txn.Epoch == intent.Txn.Epoch` at this point.

**Review by andreimatei — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32541)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/storage/engine/mvcc.go, line 2160 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32541#-LRs1xMJ7rs17YvhNELT:-LRsHNqd15yNCtoh59JA:b-qg3saj) ([raw file](https://github.com/cockroachdb/cockroach/blob/245799c44d11966371ce1f45cf90e57c85b66bc2/pkg/storage/engine/mvcc.go#L2160)):*
<details><summary><i>Previously, tbg (Tobias Schottdorf) wrote…</i></summary><blockquote>

Note the condition above this block. I think the comment makes sense, though perhaps it should be moved up to the `return false, nil`. The comment explains why we don't necessarily have `meta.Txn.Epoch == intent.Txn.Epoch` at this point.
</blockquote></details>

Well now that I read the block above, I don't quite understand it. Why is there "nothing to do if the epoch..."?
I spent time staring at this function and I can't really follow its flow. What case are we even handling around here?
AFAICS, this function deals with 3 possible ways of "resolving" an intent: committing it, aborting(deleting) it or pushing it. A push is, I guess, indicated by `intent.Status==PENDING`. But pushes also seem to be handled way above.

So - what exactly is that block above about?
And since it deals with pushes, does the comment I'm deleting here which is about aborts belong there?

**Review by tbg — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32541)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/storage/engine/mvcc.go, line 2160 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32541#-LRs1xMJ7rs17YvhNELT:-LRuuxJ5Bm_VjEs73j5z:bdgmyfn) ([raw file](https://github.com/cockroachdb/cockroach/blob/245799c44d11966371ce1f45cf90e57c85b66bc2/pkg/storage/engine/mvcc.go#L2160)):*
You're right that this isn't really clear. The main concern here is that "deleting the intent" is the catchall case in this method, and that implies that we want to be cognizant about the cases in which we're hitting it. For example, consider the innocuous-looking "This method shouldn't be called..." block. If that weren't there, I think a "slow" ResolveIntent that attempts to push the timestamp, if arriving at an old epoch, would actually remove a newer intent and lead to transaction anomalies (did I get this right? The code is really hard to parse).

The comment you deleted is an extension of that. You may ask "why does it matter if the intent is PENDING? Why don't I generally do nothing if the incoming epoch is below the meta's epoch?"
The example gives you the answer: there's a chance you receive an abort at a lower epoch, and you should actually remove the intent in that case.

Perhaps replace the comment with a note of the form (in the above block)

> Note that the case `intent.Status == ABORTED && meta.Txn.Epoch > intent.Txn.Epoch` can occur naturally:
> (example)

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/32541#-:-LSGMo1d0k0gOOVqPAAf:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/storage/engine/mvcc.go, line 2156 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32541#-LRrxidT0k5a2e7fV5FW:-LSGLBl22E3NXXbIXULv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c404eed0e6cf8bbe1b935e2166c64eb012ca7025/pkg/storage/engine/mvcc.go#L2156)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

I'll let @tbg weigh-in on the deletion of the epoch handling portion of this comment. The fix to not mention updating the metadata timestamp LGTM.
</blockquote></details>

Done.
___
*[pkg/storage/engine/mvcc.go, line 2160 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/32541#-LRs1xMJ7rs17YvhNELT:-LSGM6DRBCVbvjkzNzwl:bsanake) ([raw file](https://github.com/cockroachdb/cockroach/blob/245799c44d11966371ce1f45cf90e57c85b66bc2/pkg/storage/engine/mvcc.go#L2160)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

You're right that this isn't really clear. The main concern here is that "deleting the intent" is the catchall case in this method, and that implies that we want to be cognizant about the cases in which we're hitting it. For example, consider the innocuous-looking "This method shouldn't be called..." block. If that weren't there, I think a "slow" ResolveIntent that attempts to push the timestamp, if arriving at an old epoch, would actually remove a newer intent and lead to transaction anomalies (did I get this right? The code is really hard to parse).

The comment you deleted is an extension of that. You may ask "why does it matter if the intent is PENDING? Why don't I generally do nothing if the incoming epoch is below the meta's epoch?"
The example gives you the answer: there's a chance you receive an abort at a lower epoch, and you should actually remove the intent in that case.

Perhaps replace the comment with a note of the form (in the above block)

> Note that the case `intent.Status == ABORTED && meta.Txn.Epoch > intent.Txn.Epoch` can occur naturally:
> (example)
</blockquote></details>

OK, I put the comment back (and moved it above).

Separately, I've remove the "This method shouldn't be called in this instance" note from the other comment above, since it seems to contradict what you're saying and also because it was making its comment really ambiguous.
